internal/pkg/queue/index/manager.go

```diff
@@ -187,6 +305,10 @@ func (im *IndexManager) Pop(host string) (id string, position uint64, size uint6
 func (im *IndexManager) Close() error {
 	im.dumpTicker.Stop()
 	im.stopChan <- struct{}{}
+	im.walStopChan <- struct{}{}
+	if im.walSyncerRunning.Load() {
+		panic("walCommitsSyncer still running")
```

Comment:
panic because a goroutine is still running is not an acceptable way of handling. this will make the other components fail to close gracefully for a simple error

Reply:
04e02e60e0002137d6ec0bcf3e81cb081f9778c6

walCommitsSyncer() now shuts down gracefully